tests/integration/test_pex_import.py

```diff
@@ -141,10 +155,49 @@ def execute_with_pex_on_pythonpath(code):
 
             import colors
             import first_party
-            
-            
+
+
             print(colors.blue("42"))
             first_party.warn("Vogon")
             """
         )
     )
+
+    # Verify bootstrap code does not leak attrs from vendored code
+    assert "no leak" == execute_with_pex_on_pythonpath(
+        dedent(
+            """\
+            import __pex__
+
+            try:
+                import attr
+                print(attr.__file__)
+            except ImportError:
+                print("no leak")
+            """
+        )
+    )
+
+    # Verify bootstrap pex code is demoted to the end of `sys.path`
+    assert pex + "/.bootstrap" == execute_with_pex_on_pythonpath(
```

Comment:
Please `os.path.join(pex, ".bootstrap")` - I'm in an ongoing port to Windows and this is one of 1000 paper cuts getting all the tests green.